### PR TITLE
CP-36340 REQ-403 global, overridable tls verification flag

### PR DESCRIPTION
**Pull request by lippirk**
- expose verification ctl file path, so that xapi can
  touch/rm this when appropriate. this is the global default

- make sure we can override the global default with a ?verify_cert
  parameter. This is mostly already plumbed

### CODE REVIEW

File: stunnel/stunnel.mli [hunk 1]
```diff
@@ -18,6 +18,8 @@ exception Stunnel_verify_error of string
 
 val crl_path : string
 
+val verify_certificates_ctrl : string
+
```
Comment (psafont) [inline stunnel/stunnel.mli:22]: Why is this needed?

Comment (lippirk) [reply stunnel/stunnel.mli:22]: So xapi knows which file to touch/rm (https://github.com/xapi-project/xen-api/pull/4331/files/6c21163963b4c5564155b17cedf154e108c28c00#diff-01e4b0b8aa12fd6279ff47e010ee9ce0aca146821b90a9f667e6f0e1b89eecf0)

Comment (lindig) [reply stunnel/stunnel.mli:22]: A better name might be `verify_cert_flag_path` or similar to indicate that this is a file path.

File: stunnel/stunnel_cache.mli [hunk 2]
```diff
@@ -26,8 +26,8 @@
     will be used, otherwise we make a fresh one. *)
 val with_connect :
   ?use_fork_exec_helper:bool ->
-  ?write_to_log:(string -> unit) -> string -> int -> bool ->
-   (Stunnel.t -> 'b) -> 'b
+  ?write_to_log:(string -> unit) ->
+  ?verify_cert:bool -> string -> int -> (Stunnel.t -> 'b) -> 'b
```
Comment (lindig) [inline stunnel/stunnel_cache.mli:30]: I understood that `verify_cert` needs to be `bool option`:
* None: use global default
* Some value: use that value

File: stunnel/stunnel.mli (hunk 1 above)
Comment (lippirk) [reply stunnel/stunnel.mli:22]: Indeed that would be a better name. I kept the name as it is to reduce churn

File: stunnel/stunnel_cache.mli (hunk 2 above)
Comment (lippirk) [reply stunnel/stunnel_cache.mli:30]: It is a bool option in disguise (this is implicit with the `?` syntactic sugar)

Comment (psafont) [reply stunnel/stunnel_cache.mli:30]: See stunnel/stunnel.ml +273
```
let must_verify_cert verify_cert =
  match verify_cert with
  | Some x -> x
  | None -> Sys.file_exists verify_certificates_ctrl
```

**Review by lindig — APPROVED** [review]

**Review by psafont — APPROVED** [review]
